client/state/sites/actions.js

```diff
@@ -80,90 +68,46 @@ export function receiveSiteUpdates( sites ) {
 }
 
 /**
- * Triggers a network request to request all visible sites
- * @returns {Function}        Action thunk
+ * Returns an action object that signals the intention to request all visible sites
+ * @param  {Function}   onSuccess function to be called once sites are requested with success
+ * @returns {Object}              Action object
  */
-export function requestSites() {
-	return ( dispatch ) => {
-		dispatch( {
-			type: SITES_REQUEST
-		} );
-
-		return wpcom.me().sites( {
-			site_visibility: 'all',
-			include_domain_only: true,
-			site_activity: 'active',
-			fields: 'ID,URL,name,capabilities,jetpack,visible,is_private,is_vip,icon,plan,jetpack_modules,single_user_site,is_multisite,options', //eslint-disable-line max-len
-			options: 'is_mapped_domain,unmapped_url,admin_url,is_redirect,is_automated_transfer,allowed_file_types,show_on_front,main_network_site,jetpack_version,software_version,default_post_format,created_at,frame_nonce,publicize_permanently_disabled,page_on_front,page_for_posts,advanced_seo_front_page_description,advanced_seo_title_formats,verification_services_codes,podcasting_archive,is_domain_only,default_sharing_status,default_likes_enabled,wordads,upgraded_filetypes_enabled,videopress_enabled,permalink_structure,gmt_offset' //eslint-disable-line max-len
-		} ).then( ( response ) => {
-			dispatch( receiveSites( response.sites ) );
-			dispatch( {
-				type: SITES_REQUEST_SUCCESS
-			} );
-		} ).catch( ( error ) => {
-			dispatch( {
-				type: SITES_REQUEST_FAILURE,
-				error
-			} );
-		} );
+export function requestSites( onSuccess ) {
+	if ( onSuccess ) {
+		return {
+			type: SITES_REQUEST,
+			meta: {
+				onSuccess
+			}
+		};
+	}
+	return {
+		type: SITES_REQUEST
 	};
 }
 
 /**
- * Returns a function which, when invoked, triggers a network request to fetch
- * a site.
+ * Returns an action object that signals the intention to fetch a site.
  *
  * @param  {Number}   siteId Site ID
- * @return {Function}        Action thunk
+ * @returns {Object}        Action object
```

Comment:
Could we align type annotations and descriptions so that they begin at the same column here? Just like you already did in the other files, which is beautiful :)